src/Stratis.Features.FederatedPeg/CollateralChecker.cs

```diff
@@ -182,6 +186,12 @@ private async Task<bool> UpdateCollateralInfoAsync(CancellationToken cancellatio
 
         public bool CheckCollateral(IFederationMember federationMember)
         {
+            if (!this.isInitialized)
```

Comment:
This seems potentially a bit messy, if connectivity to the counterchain API isn't working on startup then it seems that the consensus machinery could start up first, download a block, hit this, throw, and crash. It would be a better user experience if the node waited for the counterchain API to come up. Or am I misinterpreting the startup order?

Reply:
>if connectivity to the counterchain API isn't working on startup then it seems that the consensus machinery could start up first

No, initialization is blocked until counterchain API answers